### ISSUE TITLE
v2.0.4 errors

### ISSUE THREAD

**Issue by trying-to-function** [DengSir/tdPack2#8]
I'm getting several errors and the addon no longer functions at all for me with the latest update `v2.0.4`.

```
Date: 2021-08-12 15:11:14
ID: 1
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Addon.lua line 15:
   Usage: EmbedLibrary(addon, libname, silent, offset): 'libname' - Cannot find a library instance of "LibClass-2.0".
Debug:
   [string "=[C]"]: ?
   [string "=[C]"]: ?
   [string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua"]:187: EmbedLibrary()
   [string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua"]:169: EmbedLibraries()
   [string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua"]:136: NewAddon()
   [string "@tdPack2\Addon.lua"]:15: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 2
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\Base.lua line 10:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\Base.lua"]:10: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 3
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\ItemInfo.lua line 14:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\ItemInfo.lua"]:14: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 4
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\Item.lua line 9:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\Item.lua"]:9: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 5
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\Slot.lua line 16:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\Slot.lua"]:16: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 6
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\Group.lua line 23:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\Group.lua"]:23: in main chunk
Locals:
```

```
Date: 2021-08-12 15:11:14
ID: 7
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Object\Bag.lua line 18:
   attempt to index field 'Addon' (a nil value)
Debug:
   [string "=[C]"]: ?
   [string "@tdPack2\Object\Bag.lua"]:18: in main chunk
Locals:
```

```
Date: 2021-08-12 15:13:46
ID: 33
Error occured in: Global
Count: 1
Message: ..\AddOns\tdPack2\Button\Button.lua line 32:
   bad argument #1 to 'ipairs' (table expected, got nil)
Debug:
   [string "=[C]"]: ?
   [string "=[C]"]: ?
   [string "@tdPack2\Button\Button.lua"]:32:
      tdPack2\Button\Button.lua:28
Locals:
```

```
AddOns:
  Swatter, v1.13.6652 (SwimmingSeadragon)
  AddOnSkins, v4.56
  AdvancedInterfaceOptions, v1.6.0
  AtlasLootClassic, vv2.3.1-bcc
  AtlasLootClassicData, vv2.3.1-bcc
  AtlasLootClassicDungeonsAndRaids, vv2.3.1-bcc
  Attune, v238
  AucAdvanced, v1.13.6717 (SwimmingSeadragon)
  AucFilterBasic, v1.13.6680 (SwimmingSeadragon)
  AucStatHistogram, v1.13.6681 (SwimmingSeadragon)
  AucStatiLevel, v1.13.6705 (SwimmingSeadragon)
  AucStatPurchased, v1.13.6672 (SwimmingSeadragon)
  AucStatSimple, v1.13.6634 (SwimmingSeadragon)
  AucStatStdDev, v1.13.6635 (SwimmingSeadragon)
  AucUtilFixAH, v1.13.6608 (SwimmingSeadragon)
  AutoBiographer, v2.1.3
  AutoClam, v1.0 TBC
  AutoCrop, v1.4.1
  AutoFollow, v1.2
  BadBoy, vv2-classic
  BagBrother, v
  Bagnon, v9.1.0
  BeanCounter, v1.13.6682 (SwimmingSeadragon)
  BtWQuests, v2.10.0
  BtWQuestsTheBurningCrusade, v4.4
  BuffomatClassicTBC, v2021.7.5-6abbc1de
  CharacterNotes, v9.1.0
  DBMCore, v2.5.9
  DBMStatusBarTimers, v
  Dejunk, v9.1.0
  Details, v
  DetailsStreamer, v
  DetailsTinyThreat, v
  ElvUI, v2.11
  ElvUICustomTags, v1.09
  ElvUIRedtuzk, v1.5.1
  Enchantrix, v1.13.6708 (SwimmingSeadragon)
  EnchantrixBarker, v1.13.6697 (SwimmingSeadragon)
  ExRT, v4560
  FishingBuddy, v0.7.5.5
  GearMenu, vv3.0.1
  Gladdy, v1.19-Beta
  GTFO, v4.63
  Hemlock, v1.2.3
  HidingBar, vv2.5.3
  HKCounter, v5.0
  ItemRack, v
  LeatrixMaps, v2.5.48
  LeatrixPlus, v2.5.48
  LFGBulletinBoard, v2.59
  LunaUnitFrames, v4190
  MagicMarker, v9.0.5
  MagicMarkerData, v1.0
  MessageBarker, vv0.2.5
  MikScrollingBattleText, v5.7.152
  MissingTradeSkillsListTBC, v2.5.1
  MRT, v4560
  NeverForget, v4.3
  NovaInstanceTracker, v1.20
  NovaWorldBuffs, v2.13
  OmniBar, vv12.8
  OmniCD, v2.6.21
  Outfitter, v2.0.4
  Plater, vPlater-v337-BCC
  ProfessionsExporter, vv1.1.1
  Quartz, v3.6.6
  Questie, v6.5.1
  RatingBuster, v
  Reputable, v1.26-bcc
  SafeQueue, vv5.1
  SigmaAddOnManager, v
  SigmaProfessionFilter, v
  SigmaProfessionFilterRecipeInfo, v
  SigmaProfessionFilterAlchemy, v
  SigmaProfessionFilterBeastTraining, v
  SigmaProfessionFilterCooking, v
  SigmaProfessionFilterEnchanting, v
  SigmaProfessionFilterFirstAid, v
  SigmaProfessionFilterMining, v
  SpeedyAutoLoot, v2.1.2
  Stubby, v1.13.6614 (SwimmingSeadragon)
  SuperDuperMacro, vretail 9.0.1.2, classic 1.13.5.4
  Talented, vv210718
  TalentedData, v
  TargetNameplateIndicator, v1.38
  tdPack2, v2.0.4
  ThreatClassic2, v2.38
  TomTom, vv3.0.2
  TradeSkillMaster, vv4.11.9
  TradeSkillMasterAppHelper, vv4.0.15
  WeakAuras, v3.6.1
  WeakAurasCompanion, v1.1.0
  WhatsTraining, v3.0.0
  WIM, v3.9.1
  WIMElvUISkin, v20501.01-BBC
  BlizRuntimeLib_enUS v2.5.1.20501 <none>
  (ck=a72)
```

**Comment by Kewlie** [comment]
this addon is basicly missing LibClass-2.0 library folder simple fix copy from "tdbag2 \libs\ " copy to "tdpack2\libs\" then edit "tdpack2\libs\libs.xml" add in the line "<include file= "LibClass-2.0\LibClass-2.0.xml"/>" and load up World of Warcraft and you should greeted with a propmt about sorting rules click accept unless u have your own rules then your good to go

**Comment by DengSir** [comment]
fixed in 2.0.5
